google/cloud/aiplatform/prediction/local_model.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pathlib import Path
+from typing import Optional, Type
+
+from google.cloud import aiplatform
+
+from google.cloud.aiplatform import helpers
+from google.cloud.aiplatform.compat.types import model as gca_model_compat
+from google.cloud.aiplatform.docker_utils import build
+from google.cloud.aiplatform.prediction.handler import Handler
+from google.cloud.aiplatform.prediction.handler import PredictionHandler
+from google.cloud.aiplatform.prediction.predictor import Predictor
+from google.cloud.aiplatform.utils import prediction_utils
+
+DEFAULT_PREDICT_ROUTE = "/predict"
+DEFAULT_HEALTH_ROUTE = "/health"
+DEFAULT_HTTP_PORT = 8080
+
+
+class LocalModel:
+    """Class that represents a local model."""
+
+    def __init__(
+        self, serving_container_spec: aiplatform.gapic.ModelContainerSpec,
+    ):
+        """Creates a local model instance.
+
+        Args:
+            serving_container_spec (aiplatform.gapic.ModelContainerSpec):
+                Required. The container spec of the LocalModel instance.
+        """
+        self.serving_container_spec = serving_container_spec
+
+    @classmethod
+    def create_cpr_model(
+        cls,
+        src_dir: str,
+        output_image_uri: str,
+        predictor: Optional[Type[Predictor]] = None,
+        handler: Type[Handler] = PredictionHandler,
+        base_image: str = "python:3.7",
+        requirements_path: Optional[str] = None,
+    ) -> "LocalModel":
+        """Creates a local model from a custom predictor.
+
+        This method creates a docker image to include user-provided predictor, serializer, and
+        model server. It populates the entrypoint, entrypoint.py, right under the specified
+        directory, src_dir, if it doesn't exist and generates a Dockerfile to build the image.
```

Comment:
Can you include an example of the directory structure after the autogenerated files?

Reply:
Added.

Reply:
Looks good! Thanks!